openml/extensions/extension_interface.py

```diff
@@ -0,0 +1,90 @@
+from abc import ABC, abstractmethod
+from typing import Any, List, Optional, Tuple
+
+from openml import OpenMLTask, OpenMLFlow
+from openml.runs.trace import OpenMLTraceIteration, OpenMLRunTrace
+
+
+class Extension(ABC):
+
+    """Defines the interface to connect machine learning libraries to OpenML-Python.
+    """
+
+    ################################################################################################
+    # Abstract methods for flow serialization and de-serialization
+
+    @abstractmethod
+    def flow_to_model(self, flow: OpenMLFlow) -> Any:
+        pass
+
+    @abstractmethod
+    def model_to_flow(self, model: Any) -> OpenMLFlow:
+        pass
+
+    @abstractmethod
+    def flow_to_parameters(self, model: Any) -> List:
+        pass
+
+    @abstractmethod
+    def is_estimator(self, model: Any) -> bool:
+        pass
+
+    @abstractmethod
+    def seed_model(self, model: Any, seed: Optional[int]) -> Any:
+        pass
+
+    @abstractmethod
+    def run_model_on_fold(
+        self,
+        model: Any,
+        task: OpenMLTask,
+        rep_no: int,
+        fold_no: int,
+        sample_no: int,
+        can_measure_runtime: bool,
+        add_local_measures: bool,
+        extension: 'Extension',
+    ) -> Tuple:
+        pass
+
+    ################################################################################################
+    # Abstract methods for hyperparameter optimization
+
+    @abstractmethod
+    def is_hpo_class(self, model: Any) -> bool:
```

Comment:
What does this method do? Check whether it can generate a trace?

Reply:
Correct.